samples/multi-thread/wasm-apps/main_thread_exception.c

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2019 Intel Corporation.  All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include <stdio.h>
+#include <pthread.h>
+
+typedef struct ThreadArgs {
+    int start;
+    int length;
+} ThreadArgs;
+
+void *
+thread(void *args)
+{
+    while (1) {
+        /* When other thread (including main thread) throw exception,
+            this thread can successfully exit the dead loop */
+    }
+}
+
+int
+main()
+{
+    pthread_t tids;
+
+    if (pthread_create(&tids, NULL, thread, NULL) != 0) {
+        printf("pthread_create failed\n");
+    }
+
+    /* Trigger an exception */
+    *(int *)(-1) = 100;
```

Comment:
If the code is only supposed to compile for WASM, you might consider `__builtin_trap` instead.